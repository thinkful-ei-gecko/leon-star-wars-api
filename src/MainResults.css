#results {
  text-align: left;
}

h2 {
  text-align: center;
}

#results p {
  color: #d7bf36;
}

.resultItem {
  font-weight: bold;
  margin-right: 15px;
}

ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  line-height: 30px;
}

p#noResults {
  color: white;
  font-weight: bold;
}