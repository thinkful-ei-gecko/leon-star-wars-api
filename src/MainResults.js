import React from 'react'
import GoldDivider from './GoldDivider'
import ResultsList from './ResultsList'
import './MainResults.css'
import './Button.css'
import ErrorBoundary from './ErrorBoundary'

export default function MainResults(props) {
  return (
    <section id="results">
      <h2>Your search results:</h2>
      <GoldDivider />
      <p><span className="resultItem">Category:</span> {props.category}</p>
      <p><span className="resultItem">Search Term:</span> {props.searchTerm}</p>
      <button type="button" onClick={props.clear} className="useButton">clear</button>
      <GoldDivider />
      <ErrorBoundary>
      <ResultsList results={props.results}/>
      </ErrorBoundary>
    </section>
  );
}