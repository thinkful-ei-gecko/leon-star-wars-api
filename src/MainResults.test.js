import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MainResults from './MainResults'


describe('<MainResults />', () => {

  let categoriesList = [ 'test', 'data', 'here', 'okay'];

  it('renders empty fine without props', () => {
    const wrapper = shallow(<MainResults />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders with all its props', () => {
  const wrapper = shallow(<MainResults results={categoriesList} clear={function() {}} category='catTest' searchTerm='searchTest' />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('clicking the clear button calls proper function', () => {
    const wrapper = shallow(<MainResults clear={function() {}} />)
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});