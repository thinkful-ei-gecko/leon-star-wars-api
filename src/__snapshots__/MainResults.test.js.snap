// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainResults /> clicking the clear button calls proper function 1`] = `
<section
  id="results"
>
  <h2>
    Your search results:
  </h2>
  <GoldDivider />
  <p>
    <span
      className="resultItem"
    >
      Category:
    </span>
     
  </p>
  <p>
    <span
      className="resultItem"
    >
      Search Term:
    </span>
     
  </p>
  <button
    className="useButton"
    onClick={[Function]}
    type="button"
  >
    clear
  </button>
  <GoldDivider />
  <ErrorBoundary>
    <ResultsList />
  </ErrorBoundary>
</section>
`;

exports[`<MainResults /> renders empty fine without props 1`] = `
<section
  id="results"
>
  <h2>
    Your search results:
  </h2>
  <GoldDivider />
  <p>
    <span
      className="resultItem"
    >
      Category:
    </span>
     
  </p>
  <p>
    <span
      className="resultItem"
    >
      Search Term:
    </span>
     
  </p>
  <button
    className="useButton"
    type="button"
  >
    clear
  </button>
  <GoldDivider />
  <ErrorBoundary>
    <ResultsList />
  </ErrorBoundary>
</section>
`;

exports[`<MainResults /> renders with all its props 1`] = `
<section
  id="results"
>
  <h2>
    Your search results:
  </h2>
  <GoldDivider />
  <p>
    <span
      className="resultItem"
    >
      Category:
    </span>
     
    catTest
  </p>
  <p>
    <span
      className="resultItem"
    >
      Search Term:
    </span>
     
    searchTest
  </p>
  <button
    className="useButton"
    onClick={[Function]}
    type="button"
  >
    clear
  </button>
  <GoldDivider />
  <ErrorBoundary>
    <ResultsList
      results={
        Array [
          "test",
          "data",
          "here",
          "okay",
        ]
      }
    />
  </ErrorBoundary>
</section>
`;
